test_project/views/frontend/widget_data.py

```diff
@@ -1,17 +1,19 @@
-from lona.html import Widget, Strong, Select, HTML, Div, H2, Br
+from lona.html import Strong, Select, HTML, Div, H2, Br
 from lona.static_files import Script
 from lona.view import LonaView
 from lona._json import dumps
 
 
-class TestWidget(Widget):
+class TestWidget(Div):
     STATIC_FILES = [
         Script(name='TestWidget', path='widget_data.js'),
     ]
 
-    FRONTEND_WIDGET_CLASS = 'widget_data'
+    WIDGET = 'widget_data'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```

Comment:
It's a shame that we have to introduce some new boilerplate here. Maybe that's just a bit for the documentation :)

Reply:
Yes, i know exactly what you mean. I had multiple prototypes with something like a `get_nodes()`` method or something, but to be a full widget replacement we have to have support for custom constructors. And to call `super()` in your constructor is pythonic and unsurprising at least.
I am open to proposals though.